KMaina/Horreum
# Chore: Initial Application Setup

#### What does this PR do?
Set up the application by installing the necessary frameworks and packages.
#### Description of Task to be completed?
- Install Django and Django Rest Framework
- Install and enable Flake8 linting
- Enable static type checking using mypy
- Enable code cleanup using black and isort
- Add a Makefile
#### How should this be manually tested?
- run `git pull && git fetch` to get the latest changes
- run `git checkout ch-initial-setup`
- run a migration and check if the tables were created in the database
- start the server to confirm the app was set up properly.

#### Any background context you want to provide?
N/A
#### What are the relevant Trello stories?
[Trello job card](https://trello.com/c/jiSqOLAZ/1-project-setup)
#### Screenshots (if appropriate)
N/A
